extensions/podman/src/extension.ts

```diff
@@ -297,13 +305,16 @@ export async function updateMachines(
           ]);
           socketPath = socket;
         } else {
-          const { stdout: socket } = await extensionApi.process.exec(getPodmanCli(), [
-            'machine',
-            'inspect',
-            '--format',
-            '{{.ConnectionInfo.PodmanSocket.Path}}',
-            machineName,
-          ]);
+          const podmanMachineInfo = podmanMachinesInfo.get(machineName);
+          const { stdout: socket } = await extensionApi.process.exec(
+            getPodmanCli(),
+            ['machine', 'inspect', '--format', '{{.ConnectionInfo.PodmanSocket.Path}}', machineName],
+            {
+              env: {
+                CONTAINERS_MACHINE_PROVIDER: podmanMachineInfo?.vmType ?? '',
+              },
```

Comment:
this code is repeating too many times that I think we may miss the right call.

I guess we would need a custom exec function for podman extension adding automatically that env variable

Reply:
Done